### PR TITLE
Add __session_id__ label

### PR DESCRIPTION
**Pull request by kolesnikovae**
Please see https://github.com/grafana/pyroscope/issues/2443.

The change adds a pseudo-random `__session_id__` label to the label set. This identifier is based on the hostname: if hostname can't be accessed, a random value is generated.

At the moment the label is not supported in the backend and is discarded in distributors. Next step is to make distributors aware of "sessions" and allow configuring cardinality limit of the label.

### CODE REVIEW

**Review by Rperry2174 — APPROVED** [review]
lgtm! If we're gonna hold off on "making distributors aware of sessions" then maybe create an issue so we don't forget

**Comment by kolesnikovae** [comment]
Yep, we have one: https://github.com/grafana/pyroscope/issues/2443

**Pull request merged**
